nucypher/policy/conditions/lingo.py

```diff
@@ -116,6 +118,19 @@ def eval(self, data) -> bool:
                 f"'{self.value}' is an unprocessed context variable and is not valid "
                 f"for condition evaluation."
             )
+        if is_context_variable(self.key):
+            # programming error if we get here
+            raise RuntimeError(
+                f"'{self.key}' is an unprocessed context variable and is not valid "
+                f"for condition evaluation."
+            )
+        if self.key:
+            try:
+                data = data[self.key]
```

Comment:
Is `key` always allowed to be an `int` or `string` here i.e. if you are expecting a struct/tuple to be returned I'm not sure that `string` will work here. Perhaps I'm missing the ideal scenario.

I could be wrong, but try adding some acceptance tests. You can test using `getPolicy` tests on the SubscriptionManager contract which returns a Policy struct. See https://github.com/nucypher/nucypher/blob/development/tests/acceptance/blockchain/conditions/test_conditions.py#L316.